clients/admin-ui/src/features/system/history/helpers.tsx

```diff
@@ -120,6 +95,29 @@ export const describeSystemChange = (history: SystemHistoryResponse) => {
     }
   });
 
+  return { addedFields, removedFields, changedFields };
+};
+
+/** Creates a description of the given system history entry in the style of a commit message */
+export const describeSystemChange = (history: SystemHistoryResponse) => {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const { edited_by, before, after, created_at } = history;
+
+  let addedFields: string[] = [];
+  let removedFields: string[] = [];
+  let changedFields: string[] = [];
+
+  // if the history contains custom fields, it won't have the main fields
+  // so we can just process the custom_fields object
+  if (before.custom_fields || after.custom_fields) {
+    ({ addedFields, removedFields, changedFields } =
+      categorizeFieldModifications(before.custom_fields, after.custom_fields));
+  } else {
+    // process the main fields
+    ({ addedFields, removedFields, changedFields } =
+      categorizeFieldModifications(before, after));
```

Comment:
Most of the changes in this file are just cleanup but this is the main part to focus on for the custom field diff.